src/extensions/surveys/components/QuestionTypes.tsx

```diff
@@ -79,6 +81,35 @@ export function OpenTextQuestion({
     )
 }
 
+export function parseUserPropertiesInLink(link: string, posthog?: PostHog) {
+    if (!link || !posthog || !posthog.get_property) {
+        return link
+    }
+
+    const regex = /\{\{(.*?)\}\}/g
+    let newLink = link
+    let match: RegExpExecArray | null
+
+    // Iterate over all matches and replace them
+    while (!isNull((match = regex.exec(link)))) {
+        const placeholder = match.at(0) // The full placeholder e.g. {{property_name}}
+        const propertyNameWithPotentialWhitespace = match.at(1) // Property name is now always in match[1]
+
+        if (placeholder && propertyNameWithPotentialWhitespace) {
+            const propertyName = propertyNameWithPotentialWhitespace.trim()
+            if (propertyName !== '') {
+                const propertyValue =
+                    posthog.get_property(propertyName) ||
+                    posthog.get_property(STORED_PERSON_PROPERTIES_KEY)?.[propertyName]
```

Comment:
Is this the right thing, or is there a better way?
The API cannot return the parameter name-value since it does not know who the requester is.
We need to load this dynamically on the SDK side, but the SDK might not have all the info. E.g., user logs in device A and calls identify with all props, or user logs out and logs in again but not with all the person props again, although the person props are merged server side already.
User logs in device B and loads surveys, but all the identified props aren't cached in that device.
How would be the UX for this use case specifically?